raft/src/main/scala/zio/raft/PendingReads.scala

```diff
@@ -0,0 +1,49 @@
+package zio.raft
+
+import zio.Promise
+import java.time.Instant
+import zio.UIO
+import zio.ZIO
+import zio.raft.PendingReadEntry.PendingCommand
+import zio.raft.PendingReadEntry.PendingHeartbeat
+
+sealed trait PendingReadEntry[S]:
+  val promise: Promise[NotALeaderError, S]
+
+object PendingReadEntry:
+  case class PendingCommand[S](promise: Promise[NotALeaderError, S], enqueuedAtIndex: Index) extends PendingReadEntry[S]
+  case class PendingHeartbeat[S](promise: Promise[NotALeaderError, S], timestamp: Instant, members: Peers) extends PendingReadEntry[S]
+
+
+
+case class PendingReads[S](readsPendingCommands: List[PendingCommand[S]], readsPendingHeartbeats: List[PendingHeartbeat[S]]):
+  private def addReadPendingCommand(entry: PendingCommand[S]): PendingReads[S] =
+    if (readsPendingCommands.isEmpty || entry.enqueuedAtIndex >= readsPendingCommands.last.enqueuedAtIndex) then 
```

Comment:
this starting to get hard, because now we have two type of pending commands in the list, with index and without...
this inserting order becomes much more complicated. I would drop the attempt to order the list at the moment... and just filter it